test/07-text-custom-field-list-test.py

```diff
@@ -60,3 +61,14 @@ def test_field_analyzer_is_keyword(self):
         docs = self.db.find({"location.state": "New Hampshire"})
         assert len(docs) == 1
         assert docs[0]["user_id"] == 10
+
+    def test_escaped_field(self):
+        docs = self.db.find({"name\\.first": "name dot first"})
+        assert len(docs) == 1
+        assert docs[0]["name.first"] == "name dot first"
+
+        try:
+            self.db.find({"name.first": "name dot first"})
+            raise Exception("Should have thrown an HTTPError")
```

Comment:
Since we're only indexing "name\\.first", no index exists for "name.first" so an error should be thrown.